### ISSUE TITLE
Project general structure: Complexity

### ISSUE THREAD

**Issue by lasote** [sword-and-sorcery/sword-and-sorcery#3]
Dealing with #2  I've spent more than 10 min trying to figure out why my `ui-board-imgui-glfw-opengl3/conanfile.py` didn't get updated when pulling from git until I realized that there is another `ui-board-imgui-glfw-opengl3` in the `conanquest-fantasic-adventures`. It is very hard for me to understand the project layout and the folder names.

I understand we want a "real" project to eat our own dog food, but I think the project should start simpler and should be simplified removing 1 level of indirection, having only one simple game, not episodes, and I would like to have a repository with the name of the game, not "ui-board-imgui-gldw-opengl3". 

My point is, by far, the most time-consuming task for me with this project has been to deal with the layout of the project itself and I'm still struggling. Where to locate the workspace stuff is also very hard.

We should, as the principle we always apply, start lean, and if this game goes further, then we can think about refactoring it to be a "game factory". We should have always in mind that the purpose of this is to validate Conan features in a more real-world, but adding too much complexity at the beginning won't contribute to getting a better workbench for Conan.

About the conanfile.py that is "preparing" the assets, I don't want that initially. If I need an XML I want to have it in the "game" repository at some asserts folder and I want to point to it directly.

@memsharded feedback please.

**Comment by jgsogo** [comment]
Yes, I agree, I spent some time last day making some modifications here and this is my plan so far:
 * remove `imgui`, `glfw` and `opengl3` references and add more meaningful names. If we want to use other _technology_ to render the windows we will see how to do it in the future. At the beginning I took these names from `imgui` because I didn't know what I was going to use to draw the windows and I tested more than one.
 * remove the recommendation about workspaces and focus on packages, we need the project to run based on packages and then we can recommend using a workspace. At any point, one can create the `conanws.yml` file and build his own workspace following Conan docs.
 * about the assets, we probably need some _dynamic files_ because the path to the assets could be different depending on the paths used for the local methods, but I think this could be simplified too.

wdyt?

**Comment by lasote** [comment]
My main issue is that, without checking the files, I don't know the difference (or if they are useful for something) between: `assets-dungeon`, `conanquest-fantastic-adventures`, `the labyrinth`, `sword-and-sorcery`, `ui-board-imgui-glfw-opengl3`... Sorry for my words in advance, not trying to offend, but, too much bullshit, it is very hard to understand :rofl:

About the workspaces, I'm ok with the workspace recommendation, as it should be the way we develop the project from the start. But again I don't understand the location of the files. Anyway, we might consider to print the contents of `conanws.yml` in the readme if you think that way is more clear.

For the assets, Sure, you can still pass the board.xml path based on your computer file storage location, but a path to a file, not a path to a generated file by a recipe in a build method. Keep it simple.

I would start by removing the "conanquest fantasic adventures" and "sword and sorcery" (still don't get it) idea and simplify the repositories to have a repo with the game (what is the name?) and N for the dependencies. And, yes, calling them with meaningful names.

**Comment by jgsogo** [comment]
Ok, let me modify _everything_. This has been the result of a couple of days splitting a single project into several too-small packages. We can say it is _crap_, so it can only go up and up 😉
